sway-lib-std/src/contract_id.sw

```diff
@@ -97,6 +97,108 @@ impl ContractId {
         asm(r1: amount, r2: sub_id) {
             mint r1 r2;
         };
-        self.transfer(sha256((ContractId::from(asm() { fp: b256 }), sub_id)), amount);
+        self.transfer(AssetId::new(ContractId::from(asm() { fp: b256 }), sub_id), amount);
     }
-}
\ No newline at end of file
+}
+
+/// An AssetId is used for interacting with an asset on the network. 
+///
+/// # Additional Information
+///
+/// It is calculated by taking the sha256 hash of the originating ContractId and a SubId.
+/// ie. sha256((contract_id, sub_id)).
+///
+/// An exception is the Base Asset, which is just the ZERO_B256 AssetId.
+///
+/// The SubId is used to differentiate between different assets that are created by the same contract.
+pub struct AssetId {
```

Comment:
This should not live in `contract_id.sw`

Reply:
Rn ContractId needs AssetId and AssetId needs ContractId, cant solve the import issue without putting them in the same file. Maybe this can be turned into the contract.sw file? ContractId is in the prelude anyway so doesnt matter

Reply:
maybe something like `token_ids.sw`?

Reply:
We ought to not refer to things as `token`s and instead `asset`s. The `token.sw` should be renamed to something else. They are technically different things and people should use the correct terminology.

That being said, `token_id` is a broad name, too broad imo.
Once conditional compilation is implemented then we ought to split into 2 files for correctness.